### ISSUE TITLE
Add API to define values to be used for acquisitions

### ISSUE THREAD

**Issue by mjhaye** [sldesnoo-Delft/q1simulator#1]
This would be nice for integration testing. In my test script I have included below function. Maybe it is good to add something similarly directly into `q1simulator`.

```python
    @contextmanager
    def patch_data_source(self, source: Iterable[Iterable[Iterable[float]]]) -> Iterator[None]:
        def flatten(iterable: Iterable[Any]) -> Any:
            for it in iterable:
                if isinstance(it, Iterable) and not isinstance(it, (str, bytes)):
                    yield from flatten(it)
                else:
                    yield it

        values_iterator = iter(flatten(source))

        def add_acquisition_mock(r_obj, bins, bin_index):  # type: ignore[no-untyped-def]  # pragma: no cover
            # Code copied from q1simulator.rt_renderer.Renderer._add_acquisition.
            # Only change is that a value from the iterator is stored i.s.o. the timestamp.
            t = r_obj.time
            if bins not in r_obj.acq_bins:
                r_obj._error("ACQ INDEX INVALID")  # pylint: disable=protected-access
                return
            r_obj.acq_times[bins].append((t, bin_index))
            if bin_index >= r_obj.acq_bins[bins]:
                r_obj._error("ACQ BIN INDEX INVALID")  # pylint: disable=protected-access
            elif not r_obj.acq_buffer.add(t):
                r_obj._error("ACQ BINNING FIFO ERROR")  # pylint: disable=protected-access
            else:
                value = next(values_iterator)
                if r_obj.acq_count[bins][bin_index] == 0:
                    r_obj.acq_data[bins][bin_index] = value
                else:
                    r_obj.acq_data[bins][bin_index] += value
                r_obj.acq_count[bins][bin_index] += 1

        patchers = []
        for s in self.driver.sequencers:
            rt_renderer_obj = getattr(s, "rt_renderer")
            p = patch.object(rt_renderer_obj, "_add_acquisition", partial(add_acquisition_mock, rt_renderer_obj))
            p.start()
            patchers.append(p)
        try:
            yield
        finally:
            for p in patchers:
                p.stop()
```

**Comment by sldesnoo-Delft** [comment]
I've added method `set_acquisition_mock_data` to the sequencers of q1simulator. It provides similar functionality.

It is added to q1simulator release 0.4.1. The internal structure of rt_renderer has changed so your patch code will not function anymore with this release. rt_renderer now stores 2 values per acquisition, a value for path 0 and a value for path 1.

There are 3 options for data passed to `set_acquisition_mock_data`. The value per acquire invocation can be 1 float, 1 complex or a sequence of 2 float values. If it is 1 float value, then this value will be used for both paths. If it is a complex value, then the real value is used for path 0 and the imaginary part for path 1. If it is a pair of 2 floats then obviously the first is used for path 0 and the second for path 1. (See Python doc of method.)

@mjhaye I've updated your example. However, there is a small difference in behavior. In your example there is 1 iterator for all sequencers. In the code below every sequencer uses its own internal iterator to iterate through the data. 
There won't be any difference if only 1 sequencer contains acquire statements. With multiple acquiring sequencers the behavior is different.

```python
    @contextmanager
    def patch_data_source(self, source: Iterable[Iterable[Iterable[float]]]) -> Iterator[None]:
        def flatten(iterable: Iterable[Any]) -> Any:
            for it in iterable:
                if isinstance(it, Iterable) and not isinstance(it, (str, bytes)):
                    yield from flatten(it)
                else:
                    yield it

        values = flatten(source)

        for s in self.driver.sequencers:
            s.set_acquisition_mock_data([values])
        try:
            yield
        finally:
            for s in self.driver.sequencers:
                s.set_acquisition_mock_data(None)
```
